src/roles/common/tasks/check-predeploy-prereq.yml

```diff
@@ -179,3 +179,25 @@
   fail:
     msg: "We couldn't find the user's SSH key file {{ user_ssh_pub_key }}. Is user_ssh_pub_key defined correctly in common.yml?"
   when: target_server_type | default('NONE') is not match ("openstack") and current_user_ssh_key.rc != 0
+
+- name: Check if vsd_license_file is present
+  include_role:
+    name: common
+    tasks_from: check-local-file-present
+  vars:
+    file_path: "{{ vsd_license_file }}"
+    success_message: "Found VSD license file"
+    fail_message: "Could not find VSD license file"
+    continue_on_error: "{{ vsd_continue_on_license_failure }}"
```

Comment:
Do we need a default for vsd_continue_on_license_failure or is the variable always guarenteed to be present?

Reply:
looks like it has a default value of false in common.json which is used in common.j2 to initialize vsd_continue_on_license_failure if it isn't defined externally. Otherwise, I assume its value is based on something passed in through Jenkins, like a boolean parameter

Reply:
Excerpt from common.j2
`# < Continue Metroae execution on VSD license failure >
# Continue Metroae execution even after license failure if this is set to true
#
{%- if vsd_continue_on_license_failure is defined %}
vsd_continue_on_license_failure: {{ vsd_continue_on_license_failure | lower }}
{%- else %}
# vsd_continue_on_license_failure: False
{%- endif %}`

Excerpt from common.json
`        "vsd_continue_on_license_failure": {
            "type":"boolean",
            "title": "Continue Metroae execution on VSD license failure",
            "default": false,
            "description": "Continue Metroae execution even after license failure if this is set to true",
            "propertyOrder": 530,
            "advanced":true
        },`

Reply:
Actually common.json is commenting out vsd_continue_on_license_failure so the variable will be missing if not defined. But I see group_vars.all.j2 is defining the variable as false by default so defaut will be provided that way.